cluster-controller/src/main/java/io/strimzi/controller/cluster/model/AbstractConfiguration.java

```diff
@@ -0,0 +1,84 @@
+package io.strimzi.controller.cluster.model;
+
+import io.strimzi.controller.cluster.operator.resource.StatefulSetDiff;
+import io.vertx.core.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Spliterators;
+
+public abstract class AbstractConfiguration {
+    private static final Logger log = LoggerFactory.getLogger(AbstractConfiguration.class.getName());
+
+    private Map<String, Object> options;
+
+    public AbstractConfiguration(String configurationFile, List<String> FORBIDDEN_OPTIONS) {
+        Map<String, Object> options = new HashMap<String, Object>();
+        String[] lines = configurationFile.split("\n");
```

Comment:
Aren't all the configs we care about just a `java.util.Properties` files? So why not use `Properties` for the serialization to and from a stream?

Reply:
Yeah, we can use Properties instead of Map. That should make it easier to get the file / load the perperties from the assembly. Good point!